src/compiler/builder.ts

```diff
@@ -730,12 +743,21 @@ namespace ts {
         const fileNameToFileId = new Map<string, number>();
         let fileIdsList: (readonly number[])[] | undefined;
         let fileNamesToFileIdListId: ESMap<string, number> | undefined;
-        const fileInfos = arrayFrom(state.fileInfos.entries(), ([key, value]) => {
+        const fileInfos = arrayFrom(state.fileInfos.entries(), ([key, value]): ProgramBuildInfoFileInfo => {
             // Ensure fileId
             const fileId = toFileId(key);
             Debug.assert(fileNames[fileId - 1] === relativeToBuildInfo(key));
             const signature = state.currentAffectedFilesSignatures && state.currentAffectedFilesSignatures.get(key);
-            return signature === undefined ? value : { version: value.version, signature, affectsGlobalScope: value.affectsGlobalScope };
+            const actualSignature = signature === undefined ? value.signature : signature;

```

Comment:
Is this `signature ?? value.signature`?

Reply:
yes sorry i kind of copied from original code so didnt clean it up later..